localflavor/by/forms.py

```diff
@@ -0,0 +1,102 @@
+from django import forms
+from django.utils.translation import ugettext_lazy as _
+
+from .by_regions import BY_REGIONS_CHOICES
+
+
+class BaseKwargsUpdatedField:
+    """
+    Abstract base class made to conform the DRY principle.
+
+    The initial_options is overrided by the dict in the subclasses and then
+    automatically passed to kwargs in the FormField.__init__() method.
+    """
+
+    initial_options = None
+
+    def __init__(self, *args, **kwargs):
+        kwargs.update(self.initial_options)
+        super().__init__(*args, **kwargs)
+
+
+class UpperValueMixin:
+    """
+    Mixin made to conform the DRY principle.
+
+    Overrides the to_python methods to return the value.upper()
+    """
+
+    def to_python(self, value):
+        value = super().to_python(value)
+        if value is None:
+            return value
+        else:
+            return value.upper()
+
+
+class BYRegionField(BaseKwargsUpdatedField, forms.TypedChoiceField):
+    """
+    A form field that ensures the input is a number representing the region.
+    """
+
+    initial_options = {
+        'choices': BY_REGIONS_CHOICES,
+        'empty_value': None
+    }
+
+
+class BYPassNumberField(BaseKwargsUpdatedField, UpperValueMixin, forms.RegexField):
+    """
+    A form field that validates its input is a pass number in a right format.
+
+    Apart from that, it tries to convert all characters to uppercase as
+    the format of belarussian pass number needs that to be right.
+
+    The valid input format is: XX1234567.
+    """
+
+    initial_options = {
+        'regex': r'[A-Z]{2}\d{7}',
+        'max_length': 9,
```

Comment:
`max_length` isn't needed as the regex will take care of checking the length. The same applies to other places with `max_length` is set with a regex that enforces the length (below).